python/vuln_details/blog_get_vuln_details.py

```diff
@@ -0,0 +1,337 @@
+    export_id = 0
+    try:
+        if len(sys.argv) == 2:
+            if sys.argv[1] == "-h":
+                print_help()
+            else:
+                export_id = int(sys.argv[1])
+    except ValueError:
+        print("Bad string to integer conversion for search ID")
+        print_help()
+    
+    # Obtain the Kenna Security API key.
+    api_key = os.getenv('KENNA_API_KEY')
+    if api_key is None:
+        print_error("Environment variable KENNA_API_KEY is non-existent")
+        sys.exit(1)
+    
+    # HTTP headers for Kenna.
+    headers = {'Accept': 'application/json',
+               'Content-Type': 'application/json; charset=utf-8',
+               'X-Risk-Token': api_key,
+               'User-Agent': 'vuln_exports/1.0.0 (Kenna Security)'}
+
+    # You might have to change this depending on your deployment.
+    base_url = "https://api.kennasecurity.com"
+    
+    # Seleted fields to be returned.
+    selected_fields = [
+        "cve_description", 
+        "cve_id",
+        "description",
+        "details", 
+        "id",
+        "scanner_score",
+        "solution",
+        "risk_meter_score", 
+    ]
+
+    # If an export ID is not defined then do an vuln export.
+    if export_id == 0:
+        export_id = request_vuln_exports(base_url, headers, selected_fields)
+    else:
+        print_info(f"Using search ID: {export_id}")
+    check_export_status(base_url, headers, export_id)
+    
+    # Create the file name based on the search ID.
+    vuln_file_name = f"vulns_{export_id}"
+    jsonl_vuln_file_name = retrieve_vuln_data(base_url, headers, export_id, vuln_file_name)
+    
+    # Count and report.
+    num_vulns = count_lines(jsonl_vuln_file_name) 
+    if num_vulns == 1:
+        print_error(f"The format of file {jsonl_vuln_file_name} is probably JSON, not JSONL")
+        sys.exit(1)
+
+    print_info(f"File: {jsonl_vuln_file_name} with {num_vulns} vulns.")
+
+    file_name_without_ext = os.path.splitext(jsonl_vuln_file_name)[0]
+    json_vuln_file_name = f"{file_name_without_ext}.json"
+    html_vuln_file_name = f"{file_name_without_ext}.html"
+
+    # Initialize the HTML Sanitizer and open the output files.
+    sanitizer = Sanitizer()  # default configuration
+    json_f = open(json_vuln_file_name, 'w')
+    html_f = open(html_vuln_file_name, 'w')
+
+    # Read the JSONL input file and write out pretty JSON and HTML files.
+    num_lines = 0
+    with open(jsonl_vuln_file_name, 'r') as jsonl_f:
+        for line_num, vuln_line in enumerate(jsonl_f):
+            vuln = convert_from_jsonl(vuln_line)
+
+            vuln_formatted = json.dumps(vuln, sort_keys=True, indent=2)
+            json_f.write(f"{vuln_formatted}\n")
+            vuln_formatted = htmlize_vuln_with_details(vuln, sanitizer)
+            html_f.write(f"{vuln_formatted}<p>")
+
+            num_lines += 1
+
+    # Close up shop and publish results.
+    json_f.close()
+    html_f.close()
+    print(f"All pau.")
```

Comment:
Not sure what "pau" means -- is this meant to be "paused"?

Reply:
It Hawai'ian for done.

Reply:
I like it. Feel free to keep it if you want :)